app/src/main/java/io/intrepid/contest/screens/contestcreation/NewContestPresenter.java

```diff
@@ -0,0 +1,59 @@
+package io.intrepid.contest.screens.contestcreation;
+
+import android.support.annotation.NonNull;
+
+import io.intrepid.contest.base.BasePresenter;
+import io.intrepid.contest.base.PresenterConfiguration;
+import io.intrepid.contest.base.TextValidatableView;
+import io.intrepid.contest.models.Contest;
+import io.intrepid.contest.models.ContestProvider;
+import timber.log.Timber;
+
+class NewContestPresenter extends BasePresenter<NewContestContract.View> implements NewContestContract.Presenter {
+    private int screenIndex = 0;
+
+    public NewContestPresenter(@NonNull NewContestContract.View view,
+                               @NonNull PresenterConfiguration configuration) {
+        super(view, configuration);
+    }
+
+    @Override
+    public void onViewCreated() {
+        super.onViewCreated();
+        initializeContestData();
+    }
+
+    private void initializeContestData() {
+        view.showContestSubmissionPage(0);
+    }
+
+    @Override
+    public void onNextButtonClicked() {
+        try {
+            TextValidatableView view = this.view.getChildEditFragment(screenIndex);
```

Comment:
May have included the "this" because I gave the variable on the left side the same exact name. Perhaps I should change this to :  TextValidatableView validatableView = view.getChildEditFragment(screenIndex);